v2/cmd/voucher_client/defaultclient.go

```diff
@@ -58,10 +77,40 @@ func getDefaultTokenSource(ctx context.Context) (oauth2.TokenSource, error) {
 	if err != nil {
 		return nil, fmt.Errorf("error creating token source: %w", err)
 	}
+
 	ts := oauth2.ReuseTokenSource(nil, &idTokenSource{TokenSource: src})
 	return ts, nil
 }
 
+func generateIDToken(ctx context.Context, tok *oauth2.Token) (string, error) {
+	clientOpts := []option.ClientOption{option.WithTokenSource(oauth2.StaticTokenSource(tok))}
+
+	// Get the serviceAccount from the current token
+	googOauth, err := googleoauth2.NewService(ctx, clientOpts...)
+	if err != nil {
+		return "", fmt.Errorf("error creating google oauth client: %w", err)
+	}
+	tokenInfo, err := googOauth.Tokeninfo().Do()
+	if err != nil {
+		return "", fmt.Errorf("error creating google oauth client: %w", err)
```

Comment:
```suggestion
		return "", fmt.Errorf("error fetching token info: %w", err)
```

Nit: differentiate from :91